### PR TITLE
feat(replay): add mobile platforms to onboarding sidebar

### LINKED ISSUE #67695: [Replay Index] Add onboarding for mobile platforms
This will be needed when we get closer to GA, but we need to add onboarding for mobile platforms.

### PR DESCRIPTION
**Pull request by michellewzhang**
closes https://github.com/getsentry/sentry/issues/67695

### android:

https://github.com/user-attachments/assets/4ba7d3cc-b4a7-4678-8fe6-5c55baed873c

### react native:

https://github.com/user-attachments/assets/bb614c5d-6b55-4a97-b1db-a38e564cf0c1

### apple:

https://github.com/user-attachments/assets/6e30f742-13e2-498f-9d2b-261ad3c9dc18

### CODE REVIEW

File: static/app/components/onboarding/gettingStartedDoc/utils/replayOnboarding.tsx
Comment (ryan953) [inline static/app/components/onboarding/gettingStartedDoc/utils/replayOnboarding.tsx:14]: are the bodies of these function swapped?

Comment (michellewzhang) [reply static/app/components/onboarding/gettingStartedDoc/utils/replayOnboarding.tsx]: oops yes! fixed

**Review by brustolin — APPROVED** [review]
Apple onboarding looks good

File: static/app/components/onboarding/gettingStartedDoc/types.ts
Comment (ArthurKnaus) [inline static/app/components/onboarding/gettingStartedDoc/types.ts:91]: Is there a reason why the mobile docs need a separate key?
Couldn't we use `replayOnboarding` as default key and additionally set `replayOnboardingJsLoader` if the platform supports the loader script? 🤔

File: static/app/components/replaysOnboarding/sidebar.tsx
Comment (ArthurKnaus) [inline static/app/components/replaysOnboarding/sidebar.tsx:196]: Could we simplify that to only check for the presence of the `replayOnboardingJsLoader` key in the doc object?
```
const hasJsLoaderOnboarding = !!docs.replayOnboardingJsLoader
```

File: static/app/gettingStartedDocs/react-native/react-native.tsx
Comment (krystofwoldrich) [inline static/app/gettingStartedDocs/react-native/react-native.tsx:90]: These values are repeated in the configuration snippet, I would leave them out here since they are not needed and will make the init snipped a bit less overwhelming.

Also it will match the Browser JS which also by default doesn't show any config options.

```suggestion
    Sentry.mobileReplayIntegration(),
```

**Comment by krystofwoldrich** [comment]
Just a two small nits, otherwise RN looks good!

File: static/app/gettingStartedDocs/react-native/react-native.tsx
Comment (michellewzhang) [reply static/app/gettingStartedDocs/react-native/react-native.tsx:90]: oh yes, good point -- i copied this from the docs so maybe we want to update it there too? 

https://docs.sentry.io/platforms/react-native/session-replay/

File: static/app/components/replaysOnboarding/sidebar.tsx
Comment (michellewzhang) [reply static/app/components/replaysOnboarding/sidebar.tsx]: unfortunately this doesn't work with the radio buttons -- selecting a diff js framework from the dropdown will break the logic  :( just tried it

**Comment by michellewzhang** [comment]
> In our other guides the `Install` step shows how to add the SDK to a project, do we want it here as well?

yup, I can add that

File: static/app/components/onboarding/gettingStartedDoc/utils/replayOnboarding.tsx
Comment (billyvg) [inline static/app/components/onboarding/gettingStartedDoc/utils/replayOnboarding.tsx:7]:
```suggestion
    'The SDK aggressively redacts all text and images. We plan to add fine controls for redacting, but currently, we just allow either on or off. Learn more about configuring Session Replay by reading the [link:configuration docs].',
```

File: static/app/gettingStartedDocs/react-native/react-native.tsx
Comment (billyvg) [inline static/app/gettingStartedDocs/react-native/react-native.tsx:81]:
```suggestion
  dsn: "${params.dsn.public}",
```

File: static/app/gettingStartedDocs/react-native/react-native.tsx
Comment (billyvg) [inline static/app/gettingStartedDocs/react-native/react-native.tsx:85]: I'm not up to date on the RN sdk but should this still be in `_experiments`?

**Review by billyvg — COMMENTED** [review]
I did not check the android/ios syntax but everything else looks good

File: static/app/gettingStartedDocs/react-native/react-native.tsx
Comment (michellewzhang) [reply static/app/gettingStartedDocs/react-native/react-native.tsx:81]: oh if this is the right syntax we should also update our docs, it's wrong there too https://docs.sentry.io/platforms/react-native/session-replay/

cc @krystofwoldrich

File: static/app/gettingStartedDocs/android/android.tsx
Comment (ryan953) [inline static/app/gettingStartedDocs/android/android.tsx:326]:
```suggestion
              label: 'XML',
```

File: static/app/gettingStartedDocs/apple/ios.tsx
Comment (ryan953) [inline static/app/gettingStartedDocs/apple/ios.tsx:654]:
```suggestion
              label: 'CocoaPods',
```

File: static/app/gettingStartedDocs/react-native/react-native.tsx
Comment (michellewzhang) [reply static/app/gettingStartedDocs/react-native/react-native.tsx]: fixed here https://github.com/getsentry/sentry-docs/pull/11192

File: static/app/gettingStartedDocs/react-native/react-native.tsx
Comment (krystofwoldrich) [reply static/app/gettingStartedDocs/react-native/react-native.tsx:85]: Yes, `_experiments` is correct.

File: static/app/gettingStartedDocs/react-native/react-native.tsx
Comment (krystofwoldrich) [reply static/app/gettingStartedDocs/react-native/react-native.tsx]: Yes, this was an oversight in the docs, thank you for fixing it there as well.

**Review by krystofwoldrich — APPROVED** [review]
RN Looks good!

**Review by brustolin — APPROVED** [review]
Apple looks good!

**Review by ArthurKnaus — APPROVED** [review]

File: static/app/gettingStartedDocs/android/android.tsx
Comment (romtsn) [inline static/app/gettingStartedDocs/android/android.tsx:333]:
```suggestion
              language: 'kotlin',
```
I think prism supports kotlin just fine

File: static/app/gettingStartedDocs/android/android.tsx
Comment (romtsn) [inline static/app/gettingStartedDocs/android/android.tsx:362]:
```suggestion
              language: 'kotlin',
```

File: static/app/gettingStartedDocs/android/android.tsx
Comment (romtsn) [inline static/app/gettingStartedDocs/android/android.tsx:380]:
```suggestion
              language: 'kotlin',
```

File: static/app/gettingStartedDocs/android/android.tsx
Comment (romtsn) [inline static/app/gettingStartedDocs/android/android.tsx:410]:
```suggestion
              language: 'kotlin',
```

File: static/app/gettingStartedDocs/android/android.tsx
Comment (michellewzhang) [reply static/app/gettingStartedDocs/android/android.tsx:362]: ah okay, didn't know what was the standard since other onboardings use `java` for the language for kotlin

File: static/app/gettingStartedDocs/android/android.tsx
Comment (romtsn) [inline static/app/gettingStartedDocs/android/android.tsx:104]: FYI: i've already merged the PR that changes the name to `onErrorSampleRate` - do we wanna wait for the new SDK release and change it now, or rather ship this and follow-up later?

File: static/app/gettingStartedDocs/android/android.tsx
Comment (romtsn) [inline static/app/gettingStartedDocs/android/android.tsx:327]:
```suggestion
  id "io.sentry.android.gradle" version "${getPackageVersion(
    params,
    'sentry.java.android.gradle-plugin',
    '4.11.0'
  )}"
```
we should be able to get the version dynamically here to avoid updating it in the future everytime 🙂

File: static/app/gettingStartedDocs/android/android.tsx
Comment (michellewzhang) [reply static/app/gettingStartedDocs/android/android.tsx:104]: hmm @romtsn let's follow up perhaps, since we need to update the docs anyway - can do both at the same time. feel free to ping me to update when the SDK release is out

File: static/app/gettingStartedDocs/android/android.tsx
Comment (romtsn) [reply static/app/gettingStartedDocs/android/android.tsx:327]: I guess this applies for the other versions in this file (also `io.sentry:sentry-android`)

File: static/app/gettingStartedDocs/android/android.tsx
Comment (romtsn) [inline static/app/gettingStartedDocs/android/android.tsx:356]:
```suggestion
    implementation 'io.sentry:sentry-android:${getPackageVersion(
    params,
    'sentry.java.android',
    '7.12.0'
  )}'
```

File: static/app/gettingStartedDocs/android/android.tsx
Comment (michellewzhang) [reply static/app/gettingStartedDocs/android/android.tsx:362]: updated those here: https://github.com/getsentry/sentry/pull/76876

**Review by romtsn — APPROVED** [review]
left a few comments, but otherwise LGTM! If we don't wanna wait for the new Android SDK version, let's merge it

**Pull request merged**
